src/nbl/asset/utils/CDerivativeMapCreator.cpp

```diff
@@ -1,154 +1,49 @@
-{
-		using Base = CFloatingPointSeparableImageFilterKernelBase<SeparateOutXAxisKernel<Kernel>>;
-
-		Kernel kernel;
-
-	public:
-		// passthrough everything
-		using value_type = typename Kernel::value_type;
-
-		_NBL_STATIC_INLINE_CONSTEXPR auto MaxChannels = Kernel::MaxChannels; // derivative map only needs 2 channels
-
-		SeparateOutXAxisKernel(Kernel&& k) : Base(k.negative_support.x, k.positive_support.x), kernel(std::move(k)) {}
-
-		NBL_DECLARE_DEFINE_CIMAGEFILTER_KERNEL_PASS_THROUGHS(Base)
-					
-		// we need to ensure to override the default behaviour of `CFloatingPointSeparableImageFilterKernelBase` which applies the weight along every axis
-		template<class PreFilter, class PostFilter>
-		struct sample_functor_t
-		{
-				sample_functor_t(const SeparateOutXAxisKernel<Kernel>* _this, PreFilter& _preFilter, PostFilter& _postFilter) :
-					_this(_this), preFilter(_preFilter), postFilter(_postFilter) {}
-
-				inline void operator()(value_type* windowSample, core::vectorSIMDf& relativePos, const core::vectorSIMDi32& globalTexelCoord, const IImageFilterKernel::UserData* userData)
-				{
-					preFilter(windowSample, relativePos, globalTexelCoord, userData);
-					auto* scale = IImageFilterKernel::ScaleFactorUserData::cast(userData);
-					for (int32_t i=0; i<MaxChannels; i++)
-					{
-						// this differs from the `CFloatingPointSeparableImageFilterKernelBase`
-						windowSample[i] *= _this->kernel.weight(relativePos.x, i);
-						if (scale)
-							windowSample[i] *= scale->factor[i];
-					}
-					postFilter(windowSample, relativePos, globalTexelCoord, userData);
-				}
-
-			private:
-				const SeparateOutXAxisKernel<Kernel>* _this;
-				PreFilter& preFilter;
-				PostFilter& postFilter;
-		};
-
-		// the method all kernels must define and overload
-		template<class PreFilter, class PostFilter>
-		inline auto create_sample_functor_t(PreFilter& preFilter, PostFilter& postFilter) const
-		{
-			return sample_functor_t<PreFilter, PostFilter>(this,preFilter,postFilter);
-		}
-};
-
-}
-
-
 template<bool isotropicNormalization>
 core::smart_refctd_ptr<ICPUImage> CDerivativeMapCreator::createDerivativeMapFromHeightMap(ICPUImage* _inImg, ISampler::E_TEXTURE_CLAMP _uwrap, ISampler::E_TEXTURE_CLAMP _vwrap, ISampler::E_TEXTURE_BORDER_COLOR _borderColor, float* out_normalizationFactor)
 {
 	using namespace asset;
 
-	using ReconstructionKernel = CGaussianImageFilterKernel<>; // or Mitchell
-	using DerivKernel_ = CDerivativeImageFilterKernel<ReconstructionKernel>;
-	using DerivKernel = MyKernel<DerivKernel_>;
-	using XDerivKernel_ = CChannelIndependentImageFilterKernel<DerivKernel, CBoxImageFilterKernel>;
-	using YDerivKernel_ = CChannelIndependentImageFilterKernel<CBoxImageFilterKernel, DerivKernel>;
-	using XDerivKernel = SeparateOutXAxisKernel<XDerivKernel_>;
-	using YDerivKernel = SeparateOutXAxisKernel<YDerivKernel_>;
+	// or Mitchell
+	using ReconstructionFunction = CWeightFunction1D<SGaussianFunction>;
```

Comment:
An additional Gaussian/Mitchell will needlessly smooth our input and attenuate high frequencies, basically we'll get the same derivative map "as if" we made it from Mip Level 1.

As you are sure that the outputted derivative map will be same resolution you can make the Reconstruction Function a Dirac.

This is because even if you made it a perfect Sinc (which doesn't do anything to the signal - in theory), it would behave like a Dirac because the phase is always 0 and all zeros of the Sinc function align with neighbouring sample locations, and the middle/center one gets a weight of 1.

Reply:
Resolved.